Sources/Leaf/LeafEncoder.swift

```diff
@@ -191,11 +180,8 @@ private final class UnkeyedContainer: UnkeyedEncodingContainer, _Container {
 
     func encode<T>(_ value: T) throws where T: Encodable {
         defer { self.count += 1 }
-        if let convertible = value as? LeafDataRepresentable {
-            guard let converted = convertible.leafData else {
-                throw EncodingError.invalidValue(value, at: self.codingPath )
-            }
-            self.array.append(.data(converted))
+        if let leafRepresentable = value as? LeafDataRepresentable {
+            self.array.append(.data(leafRepresentable.leafData))
         } else {
```

Comment:
Remove conditional assignment from change to `LeafDataRepresentable.leafData: LeafData`